subprojects/launcher/src/integTest/groovy/org/gradle/launcher/daemon/server/scaninfo/DaemonScanInfoIntegrationSpec.groovy

```diff
@@ -97,24 +97,28 @@ class DaemonScanInfoIntegrationSpec extends DaemonIntegrationSpec {
            ${imports()}
            ${registerTestExpirationStrategy()}
            ${registerExpirationListener()}
-           ${delayTask()}
+           ${waitForExpirationTask()}
         """
 
         when:
-        def delayResult = executer.withArguments('delay').run()
+        executer.withArguments('waitForExpiration').run()
 
         then:
-        delayResult.assertOutputContains('onExpirationEvent fired with: expiring daemon with TestExpirationStrategy')
+        file(EXPIRATION_EVENT).text.startsWith "onExpirationEvent fired with: expiring daemon with TestExpirationStrategy uuid:"
 
         when:
+        file(EXPIRATION_EVENT).delete()
```

Comment:
Probably better to use `GFileUtils.forceDelete()` to ensure that delete works or check the result of the `delete()` method.